### PR TITLE
Added headingRowColor to PaginatedDataTable

### PR DESCRIPTION
**Pull request by npsulav**
This PR adds the feature of headingRowColor to PaginatedDataTable as issued [here](https://github.com/flutter/flutter/issues/100790)

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/lib/src/material/paginated_data_table.dart [hunk 1]
```diff
@@ -158,6 +160,11 @@ class PaginatedDataTable extends StatefulWidget {
   ///
   /// This value is optional and defaults to 56.0 if not specified.
   final double headingRowHeight;
+  
+  /// The color of the heading row.
+  ///
+  /// This value is optional and defaults to theme primary color if not specified.
+  final double headingRowColor;
```
Comment (justinmc) [inline packages/flutter/lib/src/material/paginated_data_table.dart:167]: Should we just use the same docs from DataTable.headingRowTable?  
`{@macro flutter.material.dataTable.headingRowColor}`

**Review by justinmc — COMMENTED** [review]
Thanks for adding this, sorry it took me so long to review.  Let me know what you think of my comments below.

File: packages/flutter/lib/src/material/paginated_data_table.dart [hunk 2]
```diff
@@ -158,6 +160,11 @@ class PaginatedDataTable extends StatefulWidget {
   ///
   /// This value is optional and defaults to 56.0 if not specified.
   final double headingRowHeight;
+
```
Comment (justinmc) [inline packages/flutter/lib/src/material/paginated_data_table.dart:163]: There is an invisible trailing space here I think, according to the analyzer failure.

**Review by justinmc — COMMENTED** [review]
Missed one thing about the failing tests.

**Comment by Hixie** [comment]
@npsulav Thanks for your contribution. Unfortunately as written we can't take this patch, as per the review comments above. I'm going to close this PR since there has been no progress for a while, but please don't hesitate to file a new one if you do have time to get back to it! I've mentioned this PR in the corresponding issue in case anyone wants to take it over.

**Pull request closed without merging**